staging/src/k8s.io/apimachinery/pkg/apimachinery/announced/group_factory.go

```diff
@@ -89,6 +89,17 @@ func (gmf *GroupMetaFactory) Announce(groupFactoryRegistry APIGroupFactoryRegist
 		panic(err)
 	}
 	return gmf
+
+}
+
+// AnnounceIfNotAnnounced is the same as Announce, except that it returns nil
```

Comment:
For a binary like the apiserver, it imports both pkg/master and pkg/client, which have their own import_known_versions.go that call Install(), thus the double-registrations.

The original code doesn't suffer from double-registrations because the `Install()` is called in the init function of each API groupVersion. Though the `install` packages might be imported multiple times, the init() function is guaranteed to be run only once.